drag-and-drop.md

```diff
@@ -0,0 +1,126 @@
+# Drag-adn-Drop
+
+The proposal to provide native support Drag-and Drop opeerations:
+* drag and drop between widgets 
+* receive drag and drop URL(s) to application from a desktop or another application
+* start drag operation from Fyne application with URL(s) to another desltop application (pending external changes)
+
+## Background
+
+There are few issues opened and related to missing drag and drop functionality:
+* Add drag and drop support https://github.com/fyne-io/fyne/issues/142
+* Drag and drop to application icon on macOS https://github.com/fyne-io/fyne/issues/497
+
+Existing interface provide a minimum capabilities to programmatically start a drug operation and get notified when it complets.
+This functionality doesn't provide an interface to communicate between widgets within the application nor send/receive drag and drop with other applications.
+
+## Architecture / API
+
+Existing interface will not change to preserve compatibility with existing code.
+The proposed changes will extend the interface and give user missing capabilities.
+
+The **fyne.CanvasObject** wishing to initiate drag operation must implement **fyne.Draggable** interface. No interface changes.
+
+The **fyne.CanvasObject** wishing to receiver the drop must implement **fyne.DragReceiver** interface.
+
+```
+package fyne
+
+type DragReceiver interface {
+	DragDrop(Draggable)
+}
+```
+
+To pass information to receiver the **fyne.Draggable** can also implement **drag.Info** interface.
+
+```
+package drag
+
+type Info interface {
+	Payload() Payload
+}
+```
+
+To initiate a drag-and-drop outside the Fyne application the object implement **drag.AppDraggable** interface. This interface will allow the object prepare the **drag.Payload**.
+
+```
+package drag
+
+type AppDraggable interface {
+	// StartDrag return true when object ready to start drag-and-drop
+	StartDrag() bool
+	// Payload return Payload to attach to drag-and-drop session
+	Payload() Payload
+}
+```
+
+The initial implementations for **drag.Payload** will include **drag.PayloadString**, **drag.PayloadSliceOfString**. Other types can be added as needed and assuming the platform supports such type.
+
+```
+package drag
+
+type Payload interface {}
+
+type PayloadString string
+type PayloadSliceOfString []string
+
+```
+
+To receive the drag-and-drop from outside the Fyne application the **fyne.CanvasObject** must implement the **fyne.DragReceiver** interface.
```

Comment:
Instead of "the Fyne application" I think this instance it's "the current Fyne window" right?
This could be used to drag between multiple windows which is not currently possible.